crates/sui-types/src/base_types.rs

```diff
@@ -282,6 +283,22 @@ impl From<MultiSigPublicKey> for SuiAddress {
     }
 }
 
+pub fn sui_address_from_signature(sig: &GenericSignature) -> SuiResult<SuiAddress> {
```

Comment:
nit `impl From<&GenericSignature> for SuiAddress`

Reply:
I thought about that. But the syntax of `From` makes me feel the conversion should happen among things that are very alike. If this is unpopular opinion we can implement the trait instead

Reply:
i think `TryFrom` would be totally idiomatic in this case but I don't care very much one way or the other.

Reply:
updated